iocore/net/QUICNetVConnection.cc

```diff
@@ -250,11 +434,12 @@ QUICNetVConnection::start()
     this->_ack_frame_manager.set_max_ack_delay(params->max_ack_delay_in());
     this->_schedule_ack_manager_periodic(params->max_ack_delay_in());
   } else {
+    QUICTPConfigQCP tp_config(params, NET_VCONNECTION_OUT);
```

Comment:
I worry about life time of `params`. This is declared at beginning of this function, so it looks like this could be deleted when the scope goes out and `reconfigure()` is called.

https://github.com/apache/trafficserver/blob/4934fb23c18caeec12de7bd9891f92da6cc11b7b/mgmt/ProxyConfig.cc#L191-L196

Reply:
That is actually the reason I didn't pass `tp_config` to the constructor. The values in `tp_config` will be read and be stored into `QUICTransportParamters` in `start()` so it is safe as long as we call `start()` in the same scope.

I could copy every values in the constructor but it means we have to copy the values twice. Another option would be having `params` as a member variable of QNetVC, but I'm not sure if it makes sense.

Reply:
> The values in tp_config will be read and be stored into QUICTransportParamters in start() so it is safe as long as we call start() in the same scope.

Sounds like this rule go wrong easily. I prefer to use scoped_config inside of `start()` in this case.

> Another option would be having params as a member variable of QNetVC, but I'm not sure if it makes sense.

Life time of QUICNetVC and params could be same, so this sounds much better. Is there any concern with this option?

Reply:
`scoped_config` is used like another global scope. I don't like it. 

> Life time of QUICNetVC and params could be same, so this sounds much better. Is there any concern with this option?

I don't understand why it is not like that in the first place.

Reply:
I think increment refcount on constructor and decrement it on destructor can avoid the problem. I'm missing something?

Reply:
Talked offline. We'll have only one scoped_config as a member of QNetVC. Because it should not be created only once during a session. Otherwise different values can be read.